docs/types/value-types.rst

```diff
@@ -494,21 +495,21 @@ regardless of the type of the right (exponent) operand.
     Division on integer literals used to truncate in Solidity prior to version 0.4.0, but it now converts into a rational number, i.e. ``5 / 2`` is not equal to ``2``, but to ``2.5``.
 
 .. note::
-    Solidity has a number literal type for each rational number.
-    Integer literals and rational number literals belong to number literal types.
+    In solidity, rational numbers and integers expressed as literals are of number literal type.
     Moreover, all number literal expressions (i.e. the expressions that
-    contain only number literals and operators) belong to number literal
-    types.  So the number literal expressions ``1 + 2`` and ``2 + 1`` both
-    belong to the same number literal type for the rational number three.
+    contain only number literals and operators) are of number literal
+    type.  The number literal expressions ``1 + 2`` and ``2 + 1`` are both
+    of number literal type, just the same as ``3``. There are number literal subtypes, such
+    as those for fractional rational numbers like ``1.5``.
 
 
 .. note::
     Number literal expressions are converted into a non-literal type as soon as they are used with non-literal
-    expressions. Disregarding types, the value of the expression assigned to ``b``
-    below evaluates to an integer. Because ``a`` is of type ``uint128``, the
-    expression ``2.5 + a`` has to have a proper type, though. Since there is no common type
-    for the type of ``2.5`` and ``uint128``, the Solidity compiler does not accept
-    this code.
+    expressions. Disregarding types, the value of the expression ``2.5 + a + 0.5`` where ``a=1`` would be the
+    integer ``4``. However, ``a`` is of type ``uint128``. Therefore, an implicit conversion to a non-literal
```

Comment:
What I meant with "Disregarding types" was that the value that is assigned to `b` is always an integer, regardless of the type of `a` or of any other sub-expression in the assigned value. What we can maybe say is that the expression `2.5 + 1 + 0.5` evaluates to `4` (if you think that this is not clear to the reader).